src/pybind/mgr/dashboard/services/auth/saml2.py

```diff
@@ -0,0 +1,35 @@
+from typing import Any
+
+from .auth import BaseAuth, SSOAuth
+
+
+class Saml2(SSOAuth):
+    LOGIN_URL = 'auth/saml2/login'
+    LOGOUT_URL = 'auth/saml2/slo'
+    sso = True
+
+    class Saml2Config(BaseAuth.Config):
+        onelogin_settings: Any
+
+    def __init__(self, onelogin_settings):
+        self.onelogin_settings = onelogin_settings
+
+    def get_username_attribute(self):
+        return self.onelogin_settings['sp']['attributeConsumingService']['requestedAttributes'][0][
+            'name']
+
+    def to_dict(self) -> 'Saml2Config':
+        return {
+            'onelogin_settings': self.onelogin_settings
+        }
+
+    @classmethod
+    def from_dict(cls, s_dict: Saml2Config) -> 'Saml2':
+        try:
+            return Saml2(s_dict['onelogin_settings'])
+        except KeyError:
+            return Saml2({})
```

Comment:
One very subtle comment here. Imagine that you didn't write `Saml()` code and don't know whether it also triggers `KeyError`... In general it's good that try-except blocks catch only what's missing:
```suggestion
        try:
            config = s_dict['onelogin_settings'])
        except KeyError:
            config = {}
        finally:
            return Saml2(config)
```

An alternative approach (less explicit):
```suggestion
        config = s_dict.get('onelogin_settings', {})
        return Saml2(config)
```

Reply:
Makes sense. Updated it. Thanks!